src/generators/minecraftArmor/minecraftArmorGenerator.ts

```diff
@@ -0,0 +1,1634 @@
+    yOffset: glinty,
+    angle: glintr,
+  };
+
+  function getTint(colorId: string): Blend {
+    generator.defineSelectInput(colorId, [
+      "Leather",
+      "Black",
+      "Red",
+      "Green",
+      "Brown",
+      "Blue",
+      "Purple",
+      "Cyan",
+      "Light Gray",
+      "Gray",
+      "Pink",
+      "Lime",
+      "Yellow",
+      "Light Blue",
+      "Magenta",
+      "Orange",
+      "White",
+    ]);
+
+    const hex = (() => {
+      switch (generator.getSelectInputValue(colorId)) {
+        case "Leather":
+          return "A06540";
+        case "Black":
+          return "1D1D21";
+        case "Red":
+          return "B02E26";
+        case "Green":
+          return "5E7C16";
+        case "Brown":
+          return "835432";
+        case "Blue":
+          return "3C44AA";
+        case "Purple":
+          return "8932B8";
+        case "Cyan":
+          return "169C9C";
+        case "Light Gray":
+          return "9D9D97";
+        case "Gray":
+          return "474F52";
+        case "Pink":
+          return "F38BAA";
+        case "Lime":
+          return "80C71F";
+        case "Yellow":
+          return "FED83D";
+        case "Light Blue":
+          return "3AB3DA";
+        case "Magenta":
+          return "C74EBD";
+        case "Orange":
+          return "F9801D";
+        case "White":
+          return "F9FFFE";
+        default:
+          return "A06540";
+      }
+    })();
+
+    return { kind: "MultiplyHex", hex: hex };
+  }
+
+  function getPalette(id: string, length: number): Color[] {
+    return Array.from({ length }, (_, i) => {
+      const color = generator.getTexturePixelColor(id, [i, 0]);
+      return color ?? { r: 0, g: 0, b: 0, a: 255 };
+    });
+  }
+
+  const baseColors = getPalette("Trim Palette  ", 8);
+
+  function shoulderAlpha(textureId: string): number {
+    const color = generator.getTexturePixelColor(textureId, [51, 24]);
+    const a = color ? color.a : 33;
+    console.log(`${textureId}: ${a}`);
+    return a;
+  }
+
+  function drawHelmetHead(
+    textureId: string,
+    showHeadOverlay: boolean,
+    tint: Blend,
+    enchanted: boolean
+  ) {
+    const ox = 41;
+    const oy = 21;
+    const dimensions: Dimensions = [80, 80, 80];
+    const glint = enchanted
+      ? { ...glintOptions, textureId: "Enchanted Glint" }
+      : undefined;
+    minecraftGenerator.drawCuboid(
+      textureId,
+      char.base.head,
+      [ox, oy],
+      dimensions,
+      { blend: tint, glint: glint }
+    );
+    if (showHeadOverlay) {
+      minecraftGenerator.drawCuboid(
+        textureId,
+        char.overlay.head,
+        [ox, oy],
+        dimensions,
+        { blend: tint, glint: glint }
+      );
+    }
+  }
+
+  function drawHelmetLiner(
+    textureId: string,
+    showHeadOverlay: boolean,
+    tint: Blend,
+    enchanted: boolean
+  ) {
+    const ox = 329;
+    const oy = 37;
+    const dimensions: Dimensions = [64, 64, 64];
+    const glint = enchanted
```

Comment:
It's a good idea to add the type `glint: Glint` to all of these, then TypeScript will tell you if there are invalid properties such as the `textureId`